dan_layer/consensus/src/hotstuff/on_receive_proposal.rs

```diff
@@ -133,30 +166,48 @@ where
             Ok::<_, HotStuffError>(())
         })?;
 
+        if self.block_has_missing_transaction(&local_committee, &block).await? {
+            Ok(())
+        } else {
+            self.process_block(&local_committee, &block).await
+        }
+    }
+
+    pub async fn reprocess_block(&self, block_id: &BlockId) -> Result<(), HotStuffError> {
+        let block = self.store.with_read_tx(|tx| tx.blocks_get(block_id))?;
```

Comment:
Should use the record types in the consensus layer
```suggestion
        let block = self.store.with_read_tx(|tx| Block::get(tx, block_id))?;
```